lib/observability/tracing/exporter.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracing
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/gravitational/trace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+// NewExporter returns a new exporter that is configured per the provided Config.
+func NewExporter(ctx context.Context, cfg Config) (sdktrace.SpanExporter, error) {
```

Comment:
Is there a specific reason that we have a `NewExporter` that calls `NewOTLPExporter` ? Could we just move the CheckAndSetDefaults into `NewOTLPExporter` ? Or is this just here so we can have non-OTLP exports in future ?

Reply:
I think that would be my preference, purely because I don't particularly see us adding more formats anytime soon, and it takes a few seconds to work out why we have the two seperate functions. Feels like someone might end up directly calling `NewOTLPExporter` and skipping the validation.

Reply:
Awesome, thank you.